src/main/java/teammates/ui/controller/InstructorCourseInstructorEditSaveAction.java

```diff
@@ -28,7 +29,23 @@ protected ActionResult execute() throws EntityDoesNotExistException {
         InstructorAttributes instructor = logic.getInstructorForGoogleId(courseId, account.googleId);
         gateKeeper.verifyAccessible(instructor, logic.getCourse(courseId),
                                     Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_INSTRUCTOR);
-
+        boolean isDisplayedToStudents = getRequestParamValue(Const.ParamsNames.INSTRUCTOR_IS_DISPLAYED_TO_STUDENT) != null;
+        if (!isDisplayedToStudents) {
+            List<InstructorAttributes> instructors = logic.getInstructorsForCourse(courseId);
+            InstructorAttributes instructortoEdit;
+            if (instructorId == null) {
+                instructortoEdit = logic.getInstructorForEmail(courseId, instructorEmail);
+            } else {
+                instructortoEdit = logic.getInstructorForGoogleId(courseId, instructorId);
+            }
+            isError = !logic.isAtLeastOneInstructorVisibleToStudents(instructors, instructortoEdit.email);
```

Comment:
@wkurniawan07  Right I am using instructorToEdit to get instructor email id(This mail id is used while i am looking Visibility of all the instructor and i shouldn't look the visibility of current instructor). Now i can't use instructorEmail because instructor have the right to change the email id at the time of edit so he may can change the emailid so i can't  found that email id(given by him right now) in the database.
Google id can be used in this place(It is unique and not editable) but the thing is suppose instructor A added instructor B(Which is currently not joined the course) so we don't have google id for him, in that case if Instructor A edit Visibility of instructor B, i will not have google id of him to check, that's why i used this.
One solution can be easily implemented i can just add a counter in my function(implemented in CoursesLogic.java ) and check if there are 2 instructor(1 will be current instructor) whos visibility is open then i can return true.